### ISSUE TITLE
parsing of  hard-linebreak then space inside block-quote: `>\\\n A`

### ISSUE THREAD

**Issue by mikeando** [kivikakk/comrak#213]
The markdown dingus (and my alternate implementation) says that

```
>\
 A
```

should parse as 

```
  <block_quote>
    <paragraph>
      <linebreak />
      <text>A</text>
    </paragraph>
  </block_quote>
```

And gives the HTML

```
<blockquote>
<p><br />
A</p>
</blockquote>
```

Note that there is no space before the `A`.

In contrast, `comrak` retains the space before the A, giving the HTML:

```
<blockquote>
<p><br />
 A</p>
</blockquote>
```

I think the correct behaviour is to remove the leading space as the spec contains:

> 6.7Hard line breaks
> ...
> Leading spaces at the beginning of the next line are ignored:

**Comment by mikeando** [comment]
I'm happy to see if this is going to be easy to fix if you think it is worthwhile.

**Comment by mikeando** [comment]
Seems that the issue is restricted to the `\\\n` kind of hard-linebreak. The `␣␣\n` kind seems to work fine.
Also, it is not due to the otherwise empty first line.

```
> A\
␣B
```

generates the leading space,

```
<blockquote>
<p>A<br />
␣B</p>
</blockquote>
```

while 

```
> A␣␣
␣B
```

does not generate the leading space.

```
<blockquote>
<p>A<br />
B</p>
</blockquote>
```

**Comment by mikeando** [comment]
Similarly, the non-lazy form works as expected too.

```
```
>␣A\
>␣B
```

gives no extra space

```

```
<blockquote>
<p>A<br />
B</p>
</blockquote>
```

**Comment by kivikakk** [comment]
Closed by #214 — thanks!
